internal/store/node.go

```diff
@@ -532,3 +551,44 @@ func createNodeListWatch(kubeClient clientset.Interface, ns string) cache.Lister
 		},
 	}
 }
+
+func byRole(labels ...string) aggregation.Aggregation {
```

Comment:
This might be really nit-picky or wrong, but in the "AggregateBy:" declarations, "ByLabels" starts with upper case, but "byRole" is lowercase. Shouldn't byRole have public visibility to be used in the "AggregateBy:" ?

I'll admit, even reading the code in context after I pulled the path, I do not fully understand the implementation of byRole -- I might need to grok the whole pattern of internal/store in order to fully follow the code.

I can't figure out why byRole is in node.go and the other functions like ByLabels are in aggregators.go -- I would have expected to see them all together.

Reply:
`byRole` doesn't need to be public as we only refer to it in the same file. I put it here because it really is node-specific.

The logic is:
- take all the labels starting with `node-role.kubernetes.io/` (though we expect just one, there technically may be more)
- strip the prefix
- if there are no such labels, use `<none>` as a fallback

For each role (again, only one is really expected), aggregate the metric into each individual role.

The aggregation functions are expected to return effectively an array of (metric value, labels) pairs. That array (combined across all the relevant k8s objects) is then aggregated by adding together all the values with matching labels. This means an individual metric can be a part of multiple aggregated metrics, even by the same criteria (it would be weird but it's technically valid and a nasty edge case to handle otherwise).